src/cursesdef.h

```diff
@@ -62,33 +57,33 @@ using WINDOW = void;
 class window
 {
     private:
-        WINDOW *native_window;
+        void *native_window;
 
     public:
         window() : native_window( nullptr ) { }
         template<typename T>
-        window( T *const ptr ) : native_window( static_cast<WINDOW *>( ptr ) ) {
+        window( T *const ptr ) : native_window( static_cast<void *>( ptr ) ) {
         }
         ~window() {
         }
-        template<typename T = WINDOW>
+        template<typename T = void>
         T * get() const {
             return static_cast<T *>( native_window );
         }
-        operator WINDOW *() const {
+        operator void *() const {
             return get();
         }
 };
 
 struct delwin_functor {
```

Comment:
Why not get rid of the functor by deleting `native_window` in the destructor along with prohibiting copying the `window` class?

Reply:
Good idea, I hadn't considered that. But there are some place (like `advanced_inventory_pane::window`) where it's used as plain pointer (requiring default constructor, copy constructor and so on).

I planed to add an implicit conversion from `WINDOW_PTR` to `window`, so one can use either when calling the curses functions. That would encourage the use of `WINDOW_PTR` (and I already have some changed code ready for that).

I'm probably going to change `window` to contain a shared pointer, and than get rid of `WINDOW_PTR`.